classes/TaskRunner/Upgrade/UpgradeFiles.php

```diff
@@ -92,43 +93,29 @@ public function run()
      * list files to upgrade and return it as array
      * TODO: This method needs probably to be moved in FilesystemAdapter.
      *
-     * @param string $dir
+     * @param string $pathToNewRelease
      *
      * @return array|false Number of files found, or false if param is not a folder
      */
-    protected function listFilesToUpgrade($dir)
+    protected function listFilesToUpgrade($pathToNewRelease)
     {
-        $list = [];
-        if (!is_dir($dir)) {
-            $this->logger->error($this->translator->trans('[ERROR] %s does not exist or is not a directory.', [$dir], 'Modules.Autoupgrade.Admin'));
+        if (!is_dir($pathToNewRelease)) {
+            $this->logger->error($this->translator->trans('[ERROR] %s does not exist or is not a directory.', [$pathToNewRelease], 'Modules.Autoupgrade.Admin'));
             $this->logger->info($this->translator->trans('Nothing has been extracted. It seems the unzipping step has been skipped.', [], 'Modules.Autoupgrade.Admin'));
             $this->next = 'error';
 
             return false;
         }
 
-        $allFiles = scandir($dir);
-        foreach ($allFiles as $file) {
-            $fullPath = $dir . DIRECTORY_SEPARATOR . $file;
-
-            if ($this->container->getFilesystemAdapter()->isFileSkipped(
-                $file,
-                $fullPath,
-                'upgrade',
-                $this->container->getProperty(UpgradeContainer::LATEST_PATH)
-            )) {
-                if (!in_array($file, ['.', '..'])) {
-                    $this->logger->debug($this->translator->trans('File %s is preserved', [$file], 'Modules.Autoupgrade.Admin'));
-                }
-                continue;
-            }
-            $list[] = str_replace($this->container->getProperty(UpgradeContainer::LATEST_PATH), '', $fullPath);
-            if (is_dir($fullPath)) {
-                $list = array_merge($list, $this->listFilesToUpgrade($fullPath));
-            }
-        }
+        $finder = new Finder();
+        $finder->in($pathToNewRelease)
+            ->exclude($this->container->getFileFilter()->getFilesToIgnoreOnUpgrade());
```

Comment:
Unfortunately this list makes me ignore everyfolder that has `modules`, `themes/classic` etc even though it is found at the middle of the string.
Exclude does not handle regex patterns and using `notPath()` makes the results list empty.